daskernetes/core.py

```diff
@@ -85,17 +84,15 @@ def __init__(
 
         self.namespace = namespace
         self.name = name
-        self.worker_image = worker_image
-        self.worker_labels = (worker_labels or {}).copy()
-        self.threads_per_worker = threads_per_worker
-        self.env = dict(env)
 
         # Default labels that can't be overwritten
-        self.worker_labels['org.pydata.dask/cluster-name'] = name
-        self.worker_labels['app'] = 'dask'
-        self.worker_labels['component'] = 'dask-worker'
+        worker_pod_template.metadata.labels['dask.pydata.org/cluster-name'] = name
+        worker_pod_template.metadata.labels['app'] = 'dask'
+        worker_pod_template.metadata.labels['component'] = 'dask-worker'
```

Comment:
Should we do a copy before mutating the given input?

Also, should we expect a Pod or a PodSpec?  Do we want the user to specify a an ObjMeta?

Reply:
Yeah, I think we should accept Pod objects than PodSpec, since that's what all the k8s docs generally specify. And having users be able to do annotations / labels is also quite useful.